### PR TITLE
Remove webocket native dependency.

### PR DESCRIPTION
**Pull request by nebojsa94**
## Description

This PR removes websocket dependency and instead uses ws since it doesn't require native dependency compiling.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] I have selected the correct base branch.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [X] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [X] I ran `npm run test:unit` with success.
- [X] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build-all` and tested the resulting files from `dist` folder in a browser.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.

### CODE REVIEW

File: packages/web3-providers-ws/package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "eventemitter3": "^4.0.0",
     "underscore": "1.9.1",
     "web3-core-helpers": "1.2.8",
-    "websocket": "^1.0.31"
+    "ws": "^7.3.0"
```
Comment (alcuadrado) [inline packages/web3-providers-ws/package.json]: According to the documentation, this module doesn't work in the browser.

I think we need two implementations of this provider, one for node using `ws`, and another one for the browser using the native `WebSocket` implementation. They can share most of their code.

The browser version should be declared in the package.json's `browser` field: https://docs.npmjs.com/files/package.json#browser

File: packages/web3-providers-ws/src/index.js [hunk 2]
```diff
@@ -101,7 +101,10 @@ WebsocketProvider.prototype.constructor = WebsocketProvider;
  * @returns {void}
  */
 WebsocketProvider.prototype.connect = function () {
-    this.connection = new Ws(this.url, this.protocol, undefined, this.headers, this.requestOptions, this.clientConfig);
+    // TODO: include this.requestOptions and this.clientConfig
```
Comment (cgewecke) [inline packages/web3-providers-ws/src/index.js]: This TODO is important. Relevant Web3 API docs are [here][1] plus recent additions in #3566

[1]: https://web3js.readthedocs.io/en/v1.2.9/web3.html#configuration

File: test/websocket.ganache.js [hunk 3]
```diff
@@ -64,9 +62,8 @@ describe('WebsocketProvider (ganache)', function () {
         await web3.eth.getBlockNumber();
 
         await new Promise(async function(resolve){
-            web3.currentProvider.once('error', function(err){
-                assert(err.message.includes('1012'));
-                assert(err.message.includes('restart'));
+            web3.currentProvider.once('close', function(event){
+                assert.strictEqual(event.code, 1000);
```
Comment (cgewecke) [inline test/websocket.ganache.js]: Believe this test should continue work as expected. The disconnect uses an error code. Test was added in #3486.

**Comment by cgewecke** [comment]
There's an (abandoned) PR converting to N-API in the current Websocket repo fwiw.

[theturtle/Websocket PR 302][1]

[1]: https://github.com/theturtle32/WebSocket-Node/pull/302

**Comment by cgewecke** [comment]
For background - this PR is part of [a NomicLabs led initiative][1] to purge the JS Eth stack of dependencies that require node-gyp to install. It's the source of many problems as new Node versions come out and has bad installation UX.

In some cases this is straightforward and the substitution is painless. 

In others there are tradeoffs to consider and this looks like one of them...

So far the proposed replacement's issues include:
+ Configurability not obviously equivalent.
+ Disconnection behavior not identical
+ Browser compatibility needs work.

cc: @GregTheGreek 

[1]: https://github.com/ethereum/js-organization/issues/18

**Comment by alcuadrado** [comment]
Thanks for raising these points, @cgewecke! We are now exploring fixing this in `websocket` instead. We'll report back when we have any updates.

**Comment by GregTheGreek** [comment]
This as been superseded by #3686

**Pull request closed without merging**